stdlib/Printf/src/Printf.jl

```diff
@@ -217,6 +249,38 @@ const HEX = b"0123456789ABCDEF"
     return pos
 end
 
+@inline function patch_zero(spec::Spec{T}) where {T <: Ints}
+    if spec.precision > 0
+        Spec{T}(spec.leftalign, spec.plus, spec.space, false, spec.hash, spec.width, spec.precision, spec.dynamic_width, spec.dynamic_precision)
+    else
+        spec
+    end
+end
+@inline function patch_zero(spec::Spec{T}) where {T}
+    spec
+end
+
+@inline function rmdynamic(spec::Spec{T}, args, argp) where {T}
+    width, precision = spec.width, spec.precision
+    if spec.dynamic_width
+        width = args[argp]
+        argp += 1
+    end
+    if spec.dynamic_precision
+        precision = args[argp]
+        argp += 1
+    end
+    (Spec{T}(spec.leftalign, spec.plus, spec.space, spec.zero, spec.hash, width, precision, false, false), argp)
```

Comment:
Couldn't this be combined with `patch_zero`?   e.g. replace `spec.zero` with `T isa Ints && spec.precision > 0 ? false : spec.zero`

Reply:
yes, I've done something similar, we don't need to change zero unless we have a dynamic_precision so I have it :

```
    if spec.dynamic_precision
        precision = args[argp]
        if zero && T == Int && precision > 0
            zero = false
        argp += 1
    end
    (Spec{T}(spec.leftalign, spec.plus, spec.space, zero, ...
```
(and it should be T == Int right - I'm a bit new to Julia :-)